apps/arweave/src/ar_data_sync.erl

```diff
@@ -1451,11 +1485,38 @@ get_chunk(Offset, SeekOffset, Pack, Packing, StoredPacking, StoreID) ->
 			end
 	end.
 
-%% @doc Read the chunk as well as its metadata.
-%% Response is of the format:
+get_chunk_proof(Offset, SeekOffset, StoredPacking, StoreID) ->
+	case read_chunk_with_metadata(Offset, SeekOffset, StoredPacking, StoreID, false) of
+		{error, Reason} ->
+			{error, Reason};
+		{ok, DataPath, AbsoluteOffset, TXRoot, ChunkSize, TXPath} ->
+			CheckProof =
+				case validate_fetched_chunk({AbsoluteOffset, DataPath, TXPath, TXRoot,
```

Comment:
Why do we need to validate the chunk here? 

We're reading the chunk and proof from local storage, right? So are we just double checking that the proof is correct and hasn't been corrupted since we last wrote it to disk?

Reply:
We do not strictly have to; yes, we check against corruption or validate the proofs that were copied over manually from somewhere.

Reply:
Got it. Makes sense.